service/client.hpp

```diff
@@ -22,19 +22,26 @@ namespace raven
 
     client &operator+=(raven::config_id_st id)
     {
-        config_ids_.insert(id.value());
+        current_id++;
+        config_ids_.insert({{current_id.value(), id.value()}});
         return *this;
     }
 
     client &operator-=(raven::config_id_st id)
     {
-        DLOG_F(INFO, "erasing id: %d from config: %d", id.value(), static_cast<int>(this->sock_->fileno()));
+        DLOG_F(INFO, "erasing id: %lu from config: %d", id.value(), static_cast<int>(this->sock_->fileno()));
         config_ids_.erase(id.value());
         return *this;
     }
 
+    config_id_st get_current_id() const noexcept
+    {
+        return current_id;
+    }
+
   private:
     client_ptr sock_;
-    std::unordered_set<raven::config_id_st::value_type> config_ids_;
+    raven::config_id_st current_id{0};
```

Comment:
Indeed, at first glance I didn't understand what the current_id meant, next_id seems the best to me

Reply:
actually something more like `last_id` would be better since, adding a config id increment the "curent_id" then insert the config, it make sense to call for a function afterward to get the last id stored